doc/source/gapfill.md

```diff
@@ -0,0 +1,272 @@
+:  code_prompt_show: "Show code for hypsometric binning"
+:  code_prompt_hide: "Hide code for hypsometric binning"
+
+dem_2009 = xdem.DEM(xdem.examples.get_path("longyearbyen_ref_dem"))
+dem_1990 = xdem.DEM(xdem.examples.get_path("longyearbyen_tba_dem"))
+outlines_1990 = gu.Vector(xdem.examples.get_path("longyearbyen_glacier_outlines"))
+
+ddem = xdem.dDEM(dem_2009 - dem_1990, start_time=np.datetime64("1990-08-01"), end_time=np.datetime64("2009-08-01"))
+
+ddem.data /= 2009 - 1990
+
+scott_1990 = outlines_1990.query("NAME == 'Scott Turnerbreen'")
+mask = scott_1990.create_mask(ddem)
+
+ddem_bins = xdem.volume.hypsometric_binning(ddem[mask], dem_2009[mask])
+stds = xdem.volume.hypsometric_binning(ddem[mask], dem_2009[mask], aggregation_function=np.std)
+
+plt.figure(figsize=(8, 8))
+plt.grid(zorder=0)
+plt.plot(ddem_bins["value"], ddem_bins.index.mid, linestyle="--", zorder=1)
+
+plt.barh(
+    y=ddem_bins.index.mid,
+    width=stds["value"],
+    left=ddem_bins["value"] - stds["value"] / 2,
+    height=(ddem_bins.index.left - ddem_bins.index.right) * 1,
+    zorder=2,
+    edgecolor="black",
+)
+for bin in ddem_bins.index:
+    plt.vlines(ddem_bins.loc[bin, "value"], bin.left, bin.right, color="black", zorder=3)
+
+plt.xlabel("Elevation change (m / a)")
+plt.twiny()
+plt.barh(
+    y=ddem_bins.index.mid,
+    width=ddem_bins["count"] / ddem_bins["count"].sum(),
+    left=0,
+    height=(ddem_bins.index.left - ddem_bins.index.right) * 1,
+    zorder=2,
+    alpha=0.2,
+)
+plt.xlabel("Normalized area distribution (hypsometry)")
+
+plt.ylabel("Elevation (m a.s.l.)")
+
+plt.tight_layout()
+plt.show()
+```
+
+*Caption: Hypsometric elevation change of Scott Turnerbreen on Svalbard from 1990--2009. The width of the bars indicate the standard deviation of the bin. The light blue background bars show the area distribution with elevation.*
+
+## Regional hypsometric interpolation
+
+Similarly to local hypsometric interpolation, the elevation change is assumed to be largely elevation-dependent.
+With the regional approach (often also called "global"), elevation change gradients are estimated for all glaciers in an entire region, instead of estimating one by one.
+This is advantageous in respect to areas where voids are frequent, as not even a single dDEM value has to exist on a glacier in order to reconstruct it.
+Of course, the accuracy of such an averaging is much lower than if the local hypsometric approach is used (assuming it is possible).
+
+```{code-cell} ipython3
+ddem_reghyps = ddem.interpolate(method="regional_hypsometric", reference_elevation=dem_2009, mask=glaciers_1990)
+```
+
+```{code-cell} ipython3
+:tags: [hide-input]
+:mystnb:
+:  code_prompt_show: "Show plotting code"
+:  code_prompt_hide: "Hide plotting code"
+
+ddem_reghyps = ddem.copy(new_array=ddem_reghyps)
+
+# Plot before and after
+f, ax = plt.subplots(1, 2)
+ax[0].set_title("Before regional\nhypsometric\ngap-filling")
+ddem.plot(cmap='RdYlBu', vmin=-10, vmax=10, ax=ax[0])
```

Comment:
I could not find where the issue is, but on the figure "before gap-filling" there is no gap...

Reply:
Fixed